ring.js

```diff
@@ -51,8 +57,57 @@ Ring.prototype.draw = function(context) {
   })
 }
 
-Ring.prototype.update = function(colors) {
+Ring.prototype.recolor = function(colors) {
   this.notches.forEach( function(notch, i) {
     notch.props.fill = colors[i]
   })
+}
+
+Ring.prototype.project = function(origin, targets) {
+  return targets.map(function (target) {
+    var diff = origin.difference(target)
+    var dist = origin.distance(target)
+
+    var radius = dist.position / 100
+    var angle = Math.atan2(-diff.position[1], -diff.position[0]) * 180 / Math.PI
+
+    if (angle < 90) angle += 360
+    angle = 90 - angle
+
+    var offset = origin.angle() % 360
+    if (offset < 0) offset += 360
+    angle += offset
+
+    var interp = Math.max(1 - radius, 0)
+    var fill = color.interpolateLab('rgb(55,55,55)', target.color)(interp)
+
+    return {angle: angle, radius: radius, fill: fill}
```

Comment:
This means all cues are the same size - which is ok, but if we wanted that to be different, could factor in here (and earlier)